lib/util/template/index.ts

```diff
@@ -201,15 +213,18 @@ export function compile(
   input: CompileInput,
   filterFields = true
 ): string {
-  const data = { ...GlobalConfig.get(), ...input };
+  const childEnv = {
+    env: getChildEnv({}),
+  };
+  const data = merge({}, GlobalConfig.get(), input, childEnv);
```

Comment:
```suggestion
  const data = { ...GlobalConfig.get(), ...input, env: getChildEnv({}) };
```

Reply:
Thank you for the suggestions, @viceice! I implemented all of them.

However, that brings back a bug (that Lodash `merge` was solving):

```
> input
env: {CUSTOM_FOO: 'foo'}
> childEnv
{HOME: '/root', PATH: '/tmp/yarn--...', LC_ALL: 'C.UTF-8', LANG: 'C.UTF-8'}

> { ...input, env: childEnv }
env: {HOME: '/root', PATH: '/tmp/yarn--...', LC_ALL: 'C.UTF-8', LANG: 'C.UTF-8'}

> { env: childEnv, ...input }
env: {CUSTOM_FOO: 'foo'}
```

In the example above, both `input` and `childEnv` contain key `env`. Merging then with spread operators *keeps only one `env`* (last one).

Is there a built-in method to perform a deep (recursive) merge *or* should I implement one myself?

Reply:
input shouldn't have env, as it was never supported. 🤔 why input has an env key?

lodash can't be used, as it's not a production dependency (moving type dep doesn't help) and we don't want to have that.

Reply:
The new unit tests I added use `input` to simulate custom variables that in production would be defined in `customEnvVariables` and arrive as `{ env: getChildEnv({}) }` in function `compile`.

Anyway, I found `deepmerge` used elsewhere and merged all variables this way. All tests are :green_circle: again.

Reply:
see other conmment. don't merge, don't pass env via input. simply mock `getChildEnv`, so it returns the expected values